sql: TestTxnContentionEventsTable failed

sql.TestTxnContentionEventsTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13665277?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13665277?buildTab=artifacts#/) on release-23.2 @ [f6d211271f2ba51b57d083ffc5c2cbadc38b5b23](https://github.com/cockroachdb/cockroach/commits/f6d211271f2ba51b57d083ffc5c2cbadc38b5b23):


```
=== RUN   TestTxnContentionEventsTable
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestTxnContentionEventsTable4120879167
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:146: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    crdb_internal_test.go:1032: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/crdb_internal_test.go:1032
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/crdb_internal_test.go:1051
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/crdb_internal_test.go:918
        	Error:      	"491.644574ms" is not greater than or equal to "499ms"
        	Test:       	TestTxnContentionEventsTable
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestTxnContentionEventsTable4120879167
--- FAIL: TestTxnContentionEventsTable (2.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnContentionEventsTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35633

--- a/pkg/sql/crdb_internal_test.go
+++ b/pkg/sql/crdb_internal_test.go
@@ -1004,22 +1004,35 @@ func causeContention(
 			insertValue)
 		require.NoError(t, errTxn)
 		wgTxnStarted.Done()
+		// Wait for the update to show up in cluster_queries.
+		testutils.SucceedsSoon(t, func() error {
+			row := tx.QueryRowContext(
+				ctx, "SELECT EXISTS (SELECT * FROM crdb_internal.cluster_queries WHERE query LIKE '%/* shuba */')",
+			)
+			var seen bool
+			if err := row.Scan(&seen); err != nil {
+				return err
+			}
+			if !seen {
+				return errors.Errorf("did not see update statement")
+			}
+			return nil
+		})
 		_, errTxn = tx.ExecContext(ctx, "select pg_sleep(.5);")
 		require.NoError(t, errTxn)
 		errTxn = tx.Commit()
 		require.NoError(t, errTxn)
 	}()
 
-	start := timeutil.Now()
-
 	// Need to wait for the txn to start to ensure lock contention.
 	wgTxnStarted.Wait()
-	// This will be blocked until the updateRowWithDelay finishes.
+	// This will be blocked until the insert txn finishes.
+	start := timeutil.Now()
 	_, errUpdate := conn.ExecContext(
-		ctx, fmt.Sprintf("UPDATE %s SET s = $1 where id = 'test';", table), updateValue)
+		ctx, fmt.Sprintf("UPDATE %s SET s = $1 where id = 'test' /* shuba */;", table), updateValue)
 	require.NoError(t, errUpdate)
 	end := timeutil.Now()
-	require.GreaterOrEqual(t, end.Sub(start), 499*time.Millisecond)
+	require.GreaterOrEqual(t, end.Sub(start), 500*time.Millisecond)
 
 	wgTxnDone.Wait()
 }

